handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
+}
+
+
+// ...
+this.password = new FormControl('', [
+  Validators.required,
+  hasExclamationMark
+]);
+```
+
+I've created a simple function which takes the `FormControl` instance and returns null if everything is fine.
+If the test fails, it returns an object with an arbitrarily named property.
+The property name is what will be used for the `.hasError()` test.
+
+_app/login-form.component.ts_
+```html
+<!-- ... -->
+<div [hidden]="!password.hasError('needsExclamation')">
+  Your password must have an exclamation mark!
+</div>
+<!-- ... -->
+```
+
+### Predefined Parameters
+Having a custom validator check for exclamation marks might be helpful, but what if you need to check for some other form of punctuation?
```

Comment:
I'm working on the assumption that there is less cognitive overhead to learn about a closure than to learn about developing a TypeScript interface expressly for passing in as a validator.

To wit, if we teach people that Validators are just functions, then what kind of TypeScript black-magic does a function-invocation (`Validators.minLength(5)`) return, such that the result of the function call can itself be used as a validator.
Don't we require the instantiation of a Noun to be the possessor of a verb? Is there some Angular injector we need?

No, it's really just a thunk.
There has to be some assumed level of JS proficiency, or at least higher-order function proficiency to get to the point of creating validators which hold closure references.
To that end, I would not expect junior programmers, who know nothing but Java-pre-8 to be able to create higher-order-functions nor custom validators which require them.